### PR TITLE
rephrased discovery section. Fixed #37

### LINKED ISSUE #37: Discovery inconsistency
In the [first paragraph of Discovery](https://www.w3.org/TR/pubsub/#discovery), the spec currently says that publishers MUST include an HTTP `Link` header. In the second paragraph, it says that subscribers MAY fall back to other methods of discovering the hub and topic, and suggests embedded link elements for XML and HTML documents. The inconsistency in the spec arises from the fact that if publishers are required to advertise the values in Link headers, then there is no need for consumers to check other places.

Either the spec should drop all mention of alternative methods and require publishers and consumers use HTTP headers, or it should explicitly allow publishers to advertise the values in the HTML or XML document.

We addressed this in Webmention by saying that senders are required to support both HTTP header and HTML based discovery. https://webmention.net/draft/#sender-discovers-receiver-webmention-endpoint

Personally, I would prefer not to limit discovery to only HTTP headers, since that precludes some websites from using it, so would like to do what Webmention did. Something along the lines of:
- Publishers SHOULD advertise the hub and topic in HTTP Link headers, and if the topic URL is HTML or XML, MAY instead advertise the values with a `<link>` tag.
- Subscribers MUST first check HTTP Link headers, and if the content type is HTML or XML and no values were found in the headers, then MUST check for `<link>` tags.

### PR DESCRIPTION
**Pull request by julien51**
cc @aaronpk @dret @aaronpk 
Input welcome! I believe I did not change the initially described intent, but rephrased to make it clearer. (`.host-meta` issue to be addressed in another PR!)

### CODE REVIEW

**Comment by pfefferle** [comment]
Proposal:

`<li class="warning">Finally, publishers MAY also use Web Host Metadata [[!RFC6415]] <samp>/.well-known/host-meta</samp> to include the &lt;Link&gt; element with rel="hub". However, please note that this mechanism is currently at risk and may be deprecated. </li>`

and

`<li class="warning">In the absence of both HTTP Link headers and embedded link elements, subscribers MUST look for Web Host Metadata [[!RFC6415]] <samp>/.well-known/host-meta</samp> to include the &lt;Link&gt; element with rel="hub"</li>`

If you merge this, I can file a merge request with this changes.

File: index.html [hunk 1]
```diff
@@ -135,8 +135,23 @@
...
+    </ul>
+    <p>The protocol currently supports the following discovery mechanisms. Publishers MUST implement at least one of them:</p>
+    <ul>
+      <li>Link Headers [[!RFC5988]]: the publisher SHOULD include at least one Link Header [[!RFC5988]] with <samp>rel=hub</samp> (a hub link header) as well as exactly one Link Header [[!RFC5988]] with <samp>rel=self</samp> (the self link header)</li>
+      <li>If the topic is an XML based feed, publishers SHOULD use embedded link elements as described in Appendix B of Web Linking [[!RFC5988]]. Similarly, for HTML pages, publishers SHOULD use embedded link elements as described in Appendix A of Web Linking [[!RFC5988]]</li>
+      <li class="warning">Finally, publishers MAY also use the Well-Known Uniform Resource Identifiers [[!RFC5785]] <samp>.host-meta</samp> to include the &lt;Link&gt; element with rel="hub". However, please note that this mechanism is currently at risk and may be deprecated. </li>
+    </ul>
+    <p>When perfoming discovery, subscribers MUST implement all three discovery mechanisms but should respect the following order.</p>
```
Comment (aaronpk) [inline index.html]: In order to guarantee interop, the order of these is actually strict, so how about:

> subscribers MUST implement all three discovery mechanisms in the following order

Comment (pfefferle) [inline index.html]: sounds good, but I would recommend to also fix the host-meta wording before merging.

File: index.html [hunk 2]
```diff
@@ -135,8 +135,23 @@
...
+    <ol>
+      <li>Issue a GET or HEAD request to retrieve the topic. Subscribers MUST check for HTTP Link headers first</li>
+      <li>In the absence of HTTP Link headers, and if the topic is an XML based feed or an HTML page subscribers MUST check for embedded link elements.</li>
+      <li class="warning">In the absence of both HTTP Link headers and embedded link elements, subscribers MUST look for Well-Known Uniform Resource Identifiers [[!RFC5785]] <samp>.host-meta</samp> to include the &lt;Link&gt; element with rel="hub"</li>
```
Comment (cweiske) [inline index.html]: host-meta is defined in https://tools.ietf.org/html/rfc6415#section-6.1 and not in RFC 5785

Comment (pfefferle) [inline index.html]: yes, I proposed this change above https://github.com/w3c/pubsub/pull/61#issuecomment-261702937

**Pull request merged**